warehouse/macaroons/services.py

```diff
@@ -95,11 +90,16 @@ def find_userid(self, raw_macaroon):
         if dm is None:
             return None
 
+        # This can be None if the macaroon has no associated user
+        # (e.g., an OIDC-minted macaroon).
+        if dm.user is None:
+            return None
+
```

Comment:
I wonder if `find_userid` would be better served as something like `find_identity`, which could then let the user do `identity.id` with the result.

Reply:
Yeah, I think so -- I can do this as part of #12477.